topdown/parse.go

```diff
@@ -42,6 +43,17 @@ func builtinRegoParseModule(a, b ast.Value) (ast.Value, error) {
 	return term.Value, nil
 }
 
+func registerRegoMetadataBuiltinFunction(builtin *ast.Builtin) {
+	f := func(_ BuiltinContext, _ []*ast.Term, _ func(*ast.Term) error) error {
```

Comment:
[nit]

```suggestion
	f := func(BuiltinContext, []*ast.Term, func(*ast.Term) error) error {
```

I guess we could also do
```go
RegisterBuiltinFunc(builtin.Name, func (BuiltinContext, []*ast.Term, func(*ast.Term) error) error {
    return fmt.Errorf("the %s function must only be called within the scope of a rule", builtin.Name)
})
```